src/api/entities/Identity/Portfolios.ts

```diff
@@ -1,11 +1,50 @@
-import { Identity, Namespace, NumberedPortfolio } from '~/api/entities';
+import BigNumber from 'bignumber.js';
+
+import { DefaultPortfolio, Identity, Namespace, NumberedPortfolio } from '~/api/entities';
 import { createPortfolio } from '~/api/procedures';
-import { TransactionQueue } from '~/base';
+import { PolymeshError, TransactionQueue } from '~/base';
+import { ErrorCode } from '~/types';
+import { bytesToString, numberToU64, stringToIdentityId } from '~/utils';
 
 /**
  * Handles all Portfolio related functionality on the Identity side
  */
 export class Portfolios extends Namespace<Identity> {
+  /**
+   * Retrieve a numbered Portfolio or the default Portfolio if Portfolio Id is not passed
+   *
+   * @param args.porfolioId - optional, default to the default portfolio
+   */
+  public async getPortfolio(porfolioId?: BigNumber): Promise<DefaultPortfolio | NumberedPortfolio> {
+    const {
+      context,
+      context: {
+        polymeshApi: {
+          query: { portfolio },
+        },
+      },
+    } = this;
+
+    const { did } = await context.getCurrentIdentity();
+
+    if (!porfolioId) {
+      return new DefaultPortfolio({ did }, context);
+    }
+
+    const identityId = stringToIdentityId(did, context);
+    const rawPortolioNumber = numberToU64(porfolioId, context);
+    const rawPortfolioName = await portfolio.portfolios(identityId, rawPortolioNumber);
+
+    if (!bytesToString(rawPortfolioName)) {
```

Comment:
You should be able to do `rawPortfolioName.isEmpty` here if I'm not mistaken (please test it out)